pkg/cli/userfile.go

```diff
@@ -136,6 +137,57 @@ func runUserFileList(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func uploadUserFileRecursive(conn *sqlConn, srcDir, dstDir string) error {
+	srcHasTrailingSlash := strings.HasSuffix(srcDir, "/")
+	var err error
+	srcDir, err = filepath.Abs(srcDir)
+	if err != nil {
+		return err
+	}
+	dstDir = strings.TrimSuffix(dstDir, "/")
+	// We append the last element of the (absolute) source path, i.e. the source
+	// directory name, to the destination path in the following two cases:
+	//   1. The user has not specified a destination, i.e. it is empty.
+	//   2. The source has no trailing slash (à la rsync).
+	if dstDir == "" {
+		dstDir = filepath.Base(srcDir)
+	} else if !srcHasTrailingSlash {
+		dstDir = dstDir + "/" + filepath.Base(srcDir)
+	}
+	//if dstDir == "" || !srcHasTrailingSlash {
+	//	dstDir = dstDir + "/" + filepath.Base(srcDir)
+	//}
+
+	ctx := context.Background()
+
+	// TODO(nihalpednekar): Switch to the more efficient filepath.WalkDir in
+	// Go 1.16.
+	err = filepath.Walk(srcDir,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() {
+				relativePath := strings.TrimPrefix(path, srcDir+"/")
+				fmt.Printf("uploading: %s\n", relativePath)
+
+				uploadedFile, err := uploadUserFile(ctx, conn, path, dstDir+"/"+relativePath)
+				if err != nil {
+					return err
+				}
+				fmt.Printf("successfully uploaded to %s\n", uploadedFile)
+			}
+			return nil
+		})
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("successfully uploaded all files in the subtree rooted at %s\n",
+		filepath.Base(srcDir))
```

Comment:
what do you think about printing the abs path of the srcDir? Just seeing `testDir` in the tests below seemed less informative to me. I don't feel too strongly about this so I'll leave it up to you.